frontend/app-preview/src/components/AppPreviewSubMenu/AppPreviewSubMenu.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import classes from './AppPreviewSubMenu.module.css';
+import { useTranslation } from 'react-i18next';
+import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
+import { PackagesRouter } from 'app-shared/navigation/PackagesRouter';
+import { StudioPageHeaderButton, typedLocalStorage, useMediaQuery } from '@studio/components';
+import { ArrowLeftIcon } from '@studio/icons';
+import { useInstanceIdQuery } from 'app-shared/hooks/queries';
+import { MEDIA_QUERY_MAX_WIDTH } from 'app-shared/constants';
+
+export const AppPreviewSubMenu = () => {
+  const { t } = useTranslation();
+  const shouldDisplayText = !useMediaQuery(MEDIA_QUERY_MAX_WIDTH);
+  const { org, app } = useStudioEnvironmentParams();
+  const { data: instanceId } = useInstanceIdQuery(org, app);
+
+  const packagesRouter = new PackagesRouter({ org, app });
+  const queryParams: string = `?layout=${typedLocalStorage.getItem(instanceId)}`;
+  const backToEditLink: string = `${packagesRouter.getPackageNavigationUrl('editorUiEditor')}${queryParams}`;
```

Comment:
I noticed the logic inside the component for generating the back-to-edit link. What do you think about moving it into a pure function outside the component? This would make the function more reusable and improve testability by decoupling it from the component's state and side effects. What do you think about that? 😊 

```suggestion
const getBackToEditLink = (instanceId: string): string => {
  const layout = typedLocalStorage.getItem(instanceId);
  const queryParams = `?layout=${layout}`;
  return `${packagesRouter.getPackageNavigationUrl('editorUiEditor')}${queryParams}`;
}
```

Reply:
I agree 😄 
Created a hook and refactored the code and tests to work better